sdk/core/azure-core/src/main/java/com/azure/core/http/ProxyOptions.java

```diff
@@ -216,6 +221,16 @@ private static ProxyOptions attemptToLoadAzureProxy(Configuration configuration,
         }
     }
 
+    /*
+     * Helper function that sanitizes 'NO_PROXY' into a Pattern safe string.
+     */
+    private static String sanitizeNoProxy(String noProxyString) {
+        /*
+         * The 'NO_PROXY' environment variable is expected to be delimited by ','.
+         */
+        return sanitizeNonProxyHosts(noProxyString.split(","));
+    }
```

Comment:
The names of both these methods are very similar - sanitizeNoProxy and sanitizeNonProxyHosts and they almost do the same thing. It might be worth combining them or having a more descriptive method name.

Reply:
Removed the method being called into as the expected format for `NO_PROXY` and `http.nonProxyHosts` differ enough to warrant separate code paths for sanitation.